python/cog/server/worker.py

```diff
@@ -355,26 +178,34 @@ def _loop_sync(self) -> None:
                 break
             if isinstance(ev, PredictionInput):
                 self._predict_sync(ev)
+            elif isinstance(ev, Cancel):
+                pass  # we should have gotten a signal
```

Comment:
I don't think I understand this. Are cancel events sent down the pipe or are they communicated via signal? It should be one or the other, not both, and if we're sending them down the pipe but then ignoring them... why?

Reply:
because the parent doesn't necessarily know if the predictor is async or not, we send both the event and the signal. sync predictors ignore the event, async predictors ignore the signal.